service-clients/src/main/java/com/google/cloud/pso/rag/vector/AlloyDB.java

```diff
@@ -0,0 +1,262 @@
+package com.google.cloud.pso.rag.vector;
+
+import com.google.cloud.pso.rag.common.GCPEnvironment;
+import com.google.cloud.pso.rag.common.JDBCHelper;
+import com.google.cloud.pso.rag.common.Result;
+import com.google.cloud.pso.rag.common.Result.ErrorResponse;
+import com.pgvector.PGvector;
+import com.google.cloud.pso.rag.common.Result.Success;
+import java.sql.*;
+import java.util.*;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+import static com.google.cloud.pso.rag.common.JDBCHelper.executeSQLStmtAsync;
+
+public class AlloyDB {
+
+  private AlloyDB() {}
+
+  static String searchSQLStatement(SearchRequest request) {
+    var alloyDBConfig = GCPEnvironment.config().alloyDBConfig();
+    String queriesToSearchForStr =
+        IntStream.range(0, request.queries().size())
+            .mapToObj(
+                i ->
+                    String.format(
+                        "SELECT %d internalId, '%s' id, '%s'::vector AS embedding, %d max_nn",
+                        i,
+                        request.queries().get(i).datapoint().datapointId(),
+                        request.queries().get(i).datapoint().featureVector(),
+                        request.queries().get(i).neighborCount()))
+            .collect(Collectors.joining(" UNION ALL\n"));
+
+    return String.format(
+        "WITH query_vectors AS (\n"
```

Comment:
nit, this could be loaded from resources, but for now its okay.

Reply:
thanks, not taken